src/MagnumPlugins/KtxImporter/KtxImporter.cpp

```diff
@@ -0,0 +1,553 @@
+       but we only need the ranges. */
+    const std::size_t levelIndexSize = numMipmaps * sizeof(Implementation::KtxLevel);
+    if(f->in.suffix(offset).size() < levelIndexSize) {
+        Error{} << "Trade::KtxImporter::openData(): image header too short, expected at least" << offset + levelIndexSize << "bytes but got" << data.size();
+        return;
+    }
+    const auto levelIndex = Containers::arrayCast<Implementation::KtxLevel>(f->in.slice(offset, offset + levelIndexSize));
+    for(Implementation::KtxLevel& level: levelIndex) {
+        Utility::Endianness::littleEndianInPlace(level.byteOffset, level.byteLength, level.uncompressedByteLength);
+        CORRADE_INTERNAL_ASSERT(header.supercompressionScheme != 0 || level.byteLength == level.uncompressedByteLength);
+        CORRADE_INTERNAL_ASSERT(level.uncompressedByteLength % (numFaces * numLayers) == 0);
+        if(data.size() < level.byteOffset + level.byteLength) {
+            Error{} << "Trade::KtxImporter::openData(): mip data too short, expected at least" << level.byteOffset + level.byteLength << "bytes but got" << data.size();
+            return;
+        }
+    }
+
+    f->imageData = Containers::Array<Containers::Array<File::LevelData>>{numLayers};
+
+    for(UnsignedInt layer = 0; layer != numLayers; ++layer) {
+        f->imageData[layer] = Containers::Array<File::LevelData>{numFaces * numMipmaps};
+        /* This matches the image order of DdsImporter: faces, mipmaps */
+        std::size_t currentLevel = 0;
+        for(UnsignedInt face = 0; face != numFaces; ++face) {
+            Vector3i mipSize{size};
+
+            /* Load all mipmaps for current face */
+            for(const Implementation::KtxLevel& level: levelIndex) {
+                const std::size_t length = f->compressed
+                    ? imageLength(mipSize, f->pixelFormat.compressed)
+                    : imageLength(mipSize, f->pixelFormat.uncompressed);
+
+                /* Each mipmap byte range contains tightly packed images,
+                   ordered as follows:
+                   layers, faces, (slices, rows, columns) */
+                const std::size_t imageOffset = (layer * numFaces + face) * length;
+                if(level.byteLength < imageOffset + length) {
+                    Error{} << "Trade::KtxImporter::openData(): image data too short, expected at least" << imageOffset + length << "bytes but got" << level.byteLength;
+                    return;
+                }
+
+                f->imageData[layer][currentLevel++] = {mipSize, f->in.suffix(level.byteOffset).slice(imageOffset, imageOffset + length)};
+
+                /* Shrink to next power of 2 */
+                mipSize = Math::max(mipSize >> 1, Vector3i{1});
+            }
+        }
+    }
+
+    /** @todo Read KTXorientation and flip image? What is the default? */
+    /** @todo Read KTXswizzle and apply it. Can't do that for block-compressed formats, just ignore in that case? */
```

Comment:
Hm, probably something that's worth exposing directly in image metadata, if only the (Web)GL support wasn't so bad for this feature...

I'd say expect an identity swizzle (or a swizzle that's implemented for the BGRA etc. variants) for now and fail the import otherwise.